### PR TITLE
Feature/interactive form

### PR DESCRIPTION
**Pull request by maosan132**
Hello pals, I'd really appreciate all of your suggestions to make better my third milestone for the Bookstore project:

- [ ] Make the BooksForm component stateful
- [ ] The component should store the title and the book category in its own state.
- [ ] The component should implement the handleChange method which reacts to the user typing to input a name or to changes in the category select.
- [ ] Each change to the form should modify the component’s state.
- [ ] The component should implement handleSubmit which will save the new book in the Redux store and resets the component’s state.
- [ ] Connect the REMOVE_BOOK action to the component
- [ ] Add a new table column to the BooksList component. It should render the “Remove Book” button next to each book.
- [ ] Implement the handleRemoveBook method which accepts a book as an argument and removes it from the Redux store. Don’t forget to pass that method down to the Book component as a prop.

### CODE REVIEW

File: src/containers/BooksForm.js [hunk 1]
```diff
@@ -10,25 +12,57 @@ const categories = [
...
-      {categories.map((category) => (
-        <option key={category}>{category}</option>
-      ))}
-    </select>
-    <button type="submit">Submit</button>
-  </form>
-);
+const BooksForm = () => {
+  const [book, setBook] = useState({ title: '', category: '' });
+  // const books = useSelector((store) => store.books);
```
Comment (addod19) [inline src/containers/BooksForm.js]: ### Optional
- [ ] It would be better if we remove commented piece of code. It makes the codebase messy

File: src/containers/BooksForm.js [hunk 2]
```diff
@@ -10,25 +12,57 @@ const categories = [
...
+  let category = '';
+  const dispatch = useDispatch();
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+
+    if (name === 'title') {
+      title = value;
+      setBook({ ...book, title });
+      // dispatch(action.addBook())
```
Comment (addod19) [inline src/containers/BooksForm.js]: ### Optional
- [ ] It would be better if we remove commented piece of code. It makes the codebase messy

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (addod19) [inline .vscode/settings.json]: ### Optional
- [ ] It would be better if we add files(.vscode) that are not related to the project to the `.gitignore` file. This [article](https://code.likeagirl.io/how-to-get-rid-of-ds-store-and-node-modules-in-git-repositories-d37b8a391247) can helps us get started

File: package.json [hunk 4]
```diff
@@ -39,8 +39,8 @@
     ]
   },
   "devDependencies": {
```
Comment (addod19) [inline package.json:41]: - [ ] It would be better to improve our README file in order for the README to be descriptive. Here is what makes it descriptive; 
  - [ ]  Title of the project, In our case, we can be a little specific.
  - [ ] Description about it
It would be really helpful if we use the [Microverse](https://github.com/microverseinc/readme-template) recommended template

**Review by addod19 — CHANGES_REQUESTED** [review]
### Hi @maosan132, 

Good job so far! PR is very descriptive and easy to understand and follow :fireworks: 
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Capitalized the commit messages, I realized most of the commits have been capitalized and a few others have not. But it is important to maintain a professional repository, hence stay consistent

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

__Happy coding__ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

File: src/containers/BooksForm.js (hunk 1 above)
Comment (maosan132) [reply src/containers/BooksForm.js]: Thanks for the heads up, done.

File: src/containers/BooksForm.js (hunk 2 above)
Comment (maosan132) [reply src/containers/BooksForm.js]: Thanks for the heads up, done.

File: .vscode/settings.json (hunk 3 above)
Comment (maosan132) [reply .vscode/settings.json]: Done, thanks.

File: package.json (hunk 4 above)
Comment (maosan132) [reply package.json:41]: Thanks for the heads up. Done

File: src/containers/BooksForm.js [hunk 5]
```diff
@@ -10,25 +12,55 @@ const categories = [
...
+        onChange={(e) => handleChange(e)}
+      />
+      <h4>Category:</h4>
+      <select
+        name="category"
+        value="{category}"
+        onChange={(e) => handleChange(e)}
+      >
+        {categories.map((category) => (
+          <option key={category}>{category}</option>
```
Comment (rindrajosia) [inline src/containers/BooksForm.js:58]: - [ ] [OPTIONAL] I cannot choose any category when I want to create book. The value of the select option doesn't change. Please fix it.

File: src/containers/BooksForm.js [hunk 6]
```diff
@@ -10,25 +12,55 @@ const categories = [
...
+      setBook({ ...book, category });
+    }
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    dispatch(addBook({
+      ...book,
+      id: Math.random().toString(36).substring(7),
+    }));
```
Comment (rindrajosia) [inline src/containers/BooksForm.js:40]: - [ ] [OPTIONAL]  you should `reset the STATE `once the book is created.  [screenshot from pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone3/HandleSubmit.png)

**Review by rindrajosia — APPROVED** [review]
Hi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### OPTIONAL

- [ ] I got an error when I open your APP on my local machine. It will be nice if you fix it


- [ ] I got errors related to PropTypes when I create a book. Keep in mind that PropTypes are a mechanism to ensure that components use the correct data type and pass the right data, and that components use the right type of props, and that receiving components receive the right type of props.


- [ ] .vscode is still present in your project. Please remove it.



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

File: src/containers/BooksForm.js (hunk 5 above)
Comment (maosan132) [reply src/containers/BooksForm.js:58]: It works on my end.

File: src/containers/BooksForm.js (hunk 6 above)
Comment (maosan132) [reply src/containers/BooksForm.js:40]: Thanks, I tried but didn't pull this off. Any tips about it?

File: src/containers/BooksForm.js (hunk 5 above)
Comment (maosan132) [reply src/containers/BooksForm.js:58]: For some reason, it only doesn't work for the first book created.

Comment (maosan132) [reply src/containers/BooksForm.js:58]: It only ignores the first category selected. I don't know how to fix that.
